### ISSUE TITLE
Setting Window borderless=true and resizable=false prevents application grabbing focus

### ISSUE THREAD

**Issue by awardell** [godotengine/godot#83470]
### Godot version

v4.2.beta.custom_build [b1371806a], v4.1.2.stable

### System information

Godot v4.2.beta (b1371806a) - Windows 10.0.19045 - GLES3 (Compatibility) - NVIDIA GeForce GTX 1080 Ti (NVIDIA; 31.0.15.3623) - AMD Ryzen Threadripper 1950X 16-Core Processor (32 Threads)

### Issue description

The combination of `display/window/size/borderless=true` and `display/window/size/resizable=false` in project settings will cause the application to silently fail to automatically grab focus when it starts. Calling `Window::grab_focus()` will not grab focus either. `Window::has_focus()` will lie to you and return `true`. Only actually clicking on the window will give the application focus.

Issue occurs in editor and in exported release.

Issue potentially may be platform-specific (Windows 10)

Issue may be avoided by setting `borderless` directly via GDScript in scene

### Steps to reproduce

In Project Settings:
Set display/window/size/borderless=true
Set display/window/size/resizable=false

### Minimal reproduction project

This project contains a node with a script which prints ui_up, down, left, and right when they are pressed.
Launch the game and observe that the printouts do not occur when hitting the arrow keys.
Focus the game by clicking on the screen and observe that the printouts now do occur when hitting the arrow keys.

[ReproProject.zip](https://github.com/godotengine/godot/files/12923192/ReproProject.zip)

**Comment by Achime11** [comment]
Yeah, this bug was reported as solved but it is happening on any window mode(Fullscreen, Exclusive fullscreen, or windowed) if you have resizable off. Is is an old issue that was close #80350

**Comment by k-hviid** [comment]
> Yeah, this bug was reported as solved but it is happening on Fullscreen or Exclusive fullscreen if you have resizable off. It is an old issue that was closed #80350.
> 
> PS: I am on Windows 11

Same experience here. Borderless doesn't seem to influence it.

**Comment by nzsg** [comment]
I'm experiencing this with v4.2.2.stable.mono.official [15073afe3] - if resizable is off, the window is not focused on start.

**Comment by Pixelmusement** [comment]
I've run into this issue as well using Godot v4.2.1.stable.official [b09f793f5] on Windows 10. However, I have done some experimenting and figured out a few interesting things:

1. It really is tied to having borderless AND non-resizable. For instance, running fullscreen (exclusive or not) with resizable set to true will capture focus correctly, OR if you run in a window that doesn't take up the full screen and is both borderless and not resizable the focus will not be captured correctly.
2. Using get_window().grab_focus() fails to capture the focus since Godot already thinks it has it and thus doesn't even make an attempt to re-grab it.
3. I found a semi-workaround: If you call get_window().hide() followed immediately by get_window().show() it will fix the issue regardless of anything, BUT, this also generates a "Can't change visibility of the main window" error on the hide() command. Stranger still, if you DON'T call hide() first, despite throwing an error, calling show() does not solve the problem, you MUST call hide() then show() to grab the focus successfully despite the error thrown.

**Comment by Alidonis** [comment]
Reproduced on [v4.3.stable.mono.official.77dcf97d8](https://github.com/godotengine/godot/commit/77dcf97d82cbfe4e4615475fa52ca03da645dbd8)

In fullsceen, borderless does not affect anything, only resizeable.

**Comment by p4nu** [comment]
Bug is apparently fixed at some point. This is not reproducible in 4.7.2.stable on Windows 11
